// simple_moving_average.proto
syntax = "proto3";

package simple_moving_average;

option go_package = "api-go/src/service/ms_gateway/calculate_indicator/simple_moving_average";

import "data-analysis-python/src/get_stock_data/get_stock_data.proto"; // get_stock_data.protoをインポート

service SimpleMovingAverageService {
  rpc CalculateSimpleMovingAverage (SimpleMovingAverageRequest) returns (SimpleMovingAverageResponse) {}
}

message SimpleMovingAverageRequest {
  map<string, get_stock_data.StockData> stock_data = 1; // 完全修飾名で参照
  int32 period = 2;
}

message SimpleMovingAverageResponse {
  map<string, double> moving_average = 1; // フィールド名を修正
}

// python -m grpc_tools.protoc -I src/calculate_indicator/simple_moving_average -I .. --python_out=src/calculate_indicator/simple_moving_average --grpc_python_out=src/calculate_indicator/simple_moving_average src/calculate_indicator/simple_moving_average/simple_moving_average.proto                                                       

// Go用コンパイルコマンド
// protoc --proto_path=../data-analysis-python/src/calculate_indicator/simple_moving_average --go_out=./src/service/ms_gateway/calculate_indicator/simple_moving_average --go_opt=paths=source_relative --go-grpc_out=./src/service/ms_gateway/calculate_indicator/simple_moving_average --go-grpc_opt=paths=source_relative ../data-analysis-python/src/calculate_indicator/simple_moving_average/simple_moving_average.proto